File: evd_ros_backend/evd_ros_core/src/evd_script/program/tasks.py/initialize.py
from ..task import Task
from ..machine_operations import MachineInitialize
from ..primitives import MoveUnplanned
from .open_gripper import OpenGripper


class Initialize(Task):

    '''
    Data structure methods
    '''

    def __init__(self, homeLocUuid=None, machineUuids=[], type='', name='',
                 uuid=None, parent=None, append_type=True, primitives=None, context=None):

        if primitives == None:
            primitives = []

        primitives += [MachineInitialize(id) for id in machineUuids]

        primitives += [
            MoveUnplanned(homeLocUuid,True),
            OpenGripper()
        ]

        super(Initialize,self).__init__(
            type='initialize.'+type if append_type else type,
            name=name,
            uuid=uuid,
            parent=parent,
            append_type=append_type,
            context=context,
            primitives=primitives)

File: evd_authoring_app/src/model/evdScript/program/tasks/initialize.js
import { OpenGripper } from '.';
import { NodeParser } from '../../utilityFunctions';
import { MachineInitialize } from '../machineOperations';
import { MoveUnplanned } from '../primitives';
import { Task } from '../task';


export class Initialize extends Task {

    /*
    * Data structure methods
    */

    static typeString() {
        return 'initialize.';
    }

    static fullTypeString() {
        return Task.fullTypeString() + Initialize.typeString();
    }

    constructor(homeLocUuid=null, machineUuids=[], type='', name='', uuid=null, 
                parent=null, appendType=true, primitives=null) 
    {

        if (primitives === null) {
            primitives = [];

            primitives = primitives + machineUuids.map(id => MachineInitialize(id));

            primitives = primitives + [
                MoveUnplanned(homeLocUuid,true),
                OpenGripper()
            ];
        }

        super(
            primitives,
            type,
            name,
            uuid,
            parent,
            appendType
        );
    }

    static fromDict(dct) {
        return new Initialize(
            null,
            [],
            dct.type,
            dct.name,
            dct.uuid,
            null,
            false,
            dct.primitives.map(p => NodeParser(p))
        );
    }

    static BlocklyToolbox() {
        return { type: 'initialize' };
    }

    static BlocklyBlock() {
        return { key: 'initialize', data: {
            init: function() {
                this.appendDummyInput()
                    .appendField("Initialize");
                this.appendValueInput("home-location")
                    .setCheck(null)
                    .appendField("Home Location");
                this.setPreviousStatement(true, null);
                this.setNextStatement(true, null);
                this.setColour(210);
                this.setTooltip("initialize");
                this.setHelpUrl("initialize");
            }
        }};
    }
}